src/inventory_ui.cpp

```diff
@@ -990,6 +1121,180 @@ item_location game::inv_map_splice(
     }
 }
 
+item_location game::inv_map_splice_adjacent( item_filter inv_filter, item_filter ground_filter, item_filter vehicle_filter,
```

Comment:
Oh, at first I thought it is also reading from the ground.

I'm not too sure whether it is a good feature. 

When I play cataclysm, I always find it annoying (especially for books on bookcase that I cannot move onto) to pick up the book. Then use inventory to determine whether I already have the book. (Non red means I already have 1 copy and thus will drop it.).

Thus was thinking that for me, I will prefer to be able to see all the adjacent books and pick up those in red for keeping as collection.

Another thing that annoy me as new player is to go to item and click 1 by 1 for each book to see what it did before reading it. Thus also add in the book description function so that with 1 up/down key, will be able to see the book and select what I want to read